salt/utils/decorators/__init__.py

```diff
@@ -542,8 +542,14 @@ def _set_function(self, function):
                 f_name=function.__name__))
 
         opts = self._globals.get('__opts__', '{}')
-        use_deprecated = full_name in opts.get(self.CFG_USE_DEPRECATED, list())
-        use_superseded = full_name in opts.get(self.CFG_USE_SUPERSEDED, list())
+        pillar = self._globals.get('__pillar__', '{}')
+
+        use_deprecated = full_name in opts.get(self.CFG_USE_DEPRECATED, list()) or \
+            full_name in pillar.get(self.CFG_USE_DEPRECATED, list())
+
+        use_superseded = full_name in opts.get(self.CFG_USE_SUPERSEDED, list()) or \
+            full_name in pillar.get(self.CFG_USE_SUPERSEDED, list())
```

Comment:
@garethgreenaway Please do me a favour: remove the `\` thing enclosing whole expression in `(...)` :-)